<a href="https://colab.research.google.com/github/kaleexand/DataScience-homework/blob/main/titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [175]:
import pandas as pd
from matplotlib import pyplot as plt

df_train = pd.read_csv('train.csv')
df_train

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0              1         0       3  ...   7.2500   NaN         S
1              2         1       1  ...  71.2833   C85         C
2              3         1       3  ...   7.9250   NaN         S
3              4         1       1  ...  53.1000  C123         S
4              5         0       3  ...   8.0500   NaN         S
..           ...       ...     ...  ...      ...   ...       ...
886          887         0       2  ...  13.0000   NaN         S
887          888         1       1  ...  30.0000   B42         S
888          889         0       3  ...  23.4500   NaN         S
889          890         1       1  ...  30.0000  C148         C
890          891         0       3  ...   7.7500   NaN         Q

[891 rows x 12 columns]

In [176]:
import pandas as pd
from matplotlib import pyplot as plt

df_train = pd.read_csv('train.csv')
women = df_train[(df_train['Sex'] == "female")]
men = df_train[(df_train['Sex'] == "male")]
print("Количество женщин:", women.PassengerId.count())
print("Количество мужчин:", men.PassengerId.count())


Количество женщин: 314
Количество мужчин: 577


In [177]:
# df_train.describe()
print("Средний возраст:", df_train.Age.mean())

Средний возраст: 29.69911764705882


In [178]:
# describe_fields = ["Age"]
print("Средний возраст мужчин:",df_train[df_train.Sex == "male"].Age.mean())
print("Средний возраст женщин:",df_train[df_train.Sex == "female"].Age.mean())

Средний возраст мужчин: 30.72664459161148
Средний возраст женщин: 27.915708812260537


In [184]:
women_p1= df_train[(df_train['Sex'] == "female") & (df_train['Pclass'] == 1)]
women_p2= df_train[(df_train['Sex'] == "female") & (df_train['Pclass'] == 2)]
women_p3= df_train[(df_train['Sex'] == "female") & (df_train['Pclass'] == 3)]
men_p1= df_train[(df_train['Sex'] == "male") & (df_train['Pclass'] == 1)]
men_p2= df_train[(df_train['Sex'] == "male") & (df_train['Pclass'] == 2)]
men_p3= df_train[(df_train['Sex'] == "male") & (df_train['Pclass'] == 3)]

print("Средний возраст мужчин 1 класс:",men_p1.Age.mean())
print("Средний возраст мужчин 2 класс:",men_p2.Age.mean())
print("Средний возраст мужчин 3 класс:",men_p3.Age.mean())
print("Средний возраст женщин 1 класс:",women_p1.Age.mean())
print("Средний возраст женщин 2 класс:",women_p2.Age.mean())
print("Средний возраст женщин 3 класс:",women_p3.Age.mean())



Средний возраст мужчин 1 класс: 41.28138613861386
Средний возраст мужчин 2 класс: 30.74070707070707
Средний возраст мужчин 3 класс: 26.507588932806325
Средний возраст женщин 1 класс: 34.61176470588235
Средний возраст женщин 2 класс: 28.722972972972972
Средний возраст женщин 3 класс: 21.75


In [180]:
women_no_s= df_train[(df_train['Sex'] == "female") & (df_train['Survived'] == 0)]
women_s= df_train[(df_train['Sex'] == "female") & (df_train['Survived'] == 1)]

men_no_s= df_train[(df_train['Sex'] == "male") & (df_train['Survived'] == 0)]
men_s= df_train[(df_train['Sex'] == "male") & (df_train['Survived'] == 1)]

print("Средний возраст выживших мужчин:",men_s.Age.mean())
print("Средний возраст не выживших мужчин:",men_no_s.Age.mean())
print("Средний возраст выживших женщин:",women_s.Age.mean())
print("Средний возраст не выживших  женщин:",women_no_s.Age.mean())

Средний возраст выживших мужчин: 27.276021505376345
Средний возраст не выживших мужчин: 31.618055555555557
Средний возраст выживших женщин: 28.84771573604061
Средний возраст не выживших  женщин: 25.046875


In [181]:
cnt_pas = df_train.PassengerId.count()
print("Процент выживших мужчин:",men_s.PassengerId.count()/cnt_pas,"%")
print("Процент выживших женщин:",women_s.PassengerId.count()/cnt_pas,"%")

Процент выживших мужчин: 0.122334455667789 %
Процент выживших женщин: 0.2615039281705948 %


In [182]:
p1 = df_train[(df_train['Pclass'] == 1) & (df_train['Survived'] == 1)]
p2 = df_train[(df_train['Pclass'] == 2) & (df_train['Survived'] == 1)]
p3 = df_train[(df_train['Pclass'] == 3) & (df_train['Survived'] == 1)]

print("Процент выживших относительно каждого класса")
print("Процент выживших 1 класс:",p1.PassengerId.count()/cnt_pas,"%")
print("Процент выживших 2 класс:",p2.PassengerId.count()/cnt_pas,"%")
print("Процент выживших 3 класс:",p3.PassengerId.count()/cnt_pas,"%")

Процент выживших относительно каждого класса
Процент выживших 1 класс: 0.1526374859708193 %
Процент выживших 2 класс: 0.09764309764309764 %
Процент выживших 3 класс: 0.1335578002244669 %


Проверим гипотезу в какой степени **Sex (пол)** и **Pclass (класс)**  повлияли на то, выживет конкретный человек или нет.

In [187]:
df_train.groupby(["Pclass", "Sex"])["Survived"].value_counts(normalize=True)

Pclass  Sex     Survived
1       female  1           0.968085
                0           0.031915
        male    0           0.631148
                1           0.368852
2       female  1           0.921053
                0           0.078947
        male    0           0.842593
                1           0.157407
3       female  0           0.500000
                1           0.500000
        male    0           0.864553
                1           0.135447
Name: Survived, dtype: float64

Рассмотрим по первому классу, **female** (женщина) и **Survived** выживет ли она - 1 или не выживет - 0.

Исходя из данных, если это была женщина из 1 класса, это практически 100% гарантия, что она выживет (0,968085). У мужчин из 1 класса иная статистика, можно предположить, что они спасали женщин, но можно увидеть, что выживаемость мужчин из 1 класса выше ( ~37%), чем у мужчин 2( ~16%) и 3( ~14%) классов.

Сравним женщин из всех классов. Оценка выживаемости женщин из 2 класса ~92%, из 3 класса хуже ~50%. 

Самая маленькая вероятность выживаемости около 14% была у мужчин из 3 класса. Из 100 мужчин 3 класса только 14 спасалось.
